components/transactions/RecurrencePicker.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import {Picker} from '@react-native-picker/picker';
+import { TextInput} from 'react-native-gesture-handler';
+import { Text, View } from '../Themed';
+import * as _ from 'lodash';
+
+/*
+*   Recurrence picker component for Recurring Transaction
+*
+*/
+export default function RecurrencePicker(){ 
+    let numberOfRecurrence : number = 0;
+    const weekValue = "week";
+    const monthValue = "month";
+    const yearValue = "year";
+
+    const weekday: string[] = ["Mon", "Tues", "Wed", "Thu", "Fri", "Sat", "Sun"];
+
+    var date: string[] = [];
+    for (let i = 31; i > 0; i--){
+        date.push(_.toString(i));
+    }
+    date = date.reverse();
+
+    const months: string[] = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+    const monthEndDate = [31, 30, 29];
+
```

Comment:
It seems like you're trying to reinvent DateTime picker components. Try to take a look at pre-built date-time-pickers. We shouldn't need to redefine months, dates, etc.